src/gluonts/model/evaluation.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import logging
+from collections import ChainMap
+from typing import Iterable, Optional, Union
+from dataclasses import dataclass
+from toolz import first, valmap
+
+import numpy as np
+import pandas as pd
+
+from gluonts.dataset import DataEntry
+from gluonts.dataset.split import TestData
+from gluonts.time_feature.seasonality import get_seasonality
+from gluonts.model.forecast import Forecast
+from gluonts.ev import seasonal_error
+from gluonts.itertools import prod
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class BatchForecast:
+    """
+    Wrapper around ``Forecast`` objects, that adds a batch dimension
+    to arrays returned by ``__getitem__``, for compatibility with
+    ``gluonts.ev``.
+    """
+
+    forecast: Forecast
+    allow_nan: bool = False
+
+    def __getitem__(self, name):
+        value = self.forecast[name]
+        if np.isnan(value).any():
```

Comment:
Is this actually the place to check for nan?

Why don't we dot it in get_data_batch?

Reply:
Because in `get_data_batch` I don't know what forecast keys will be accessed